.github/CODEOWNERS

```diff
@@ -167,6 +168,7 @@ receiver/cloudfoundryreceiver/                           @open-telemetry/collect
 receiver/collectdreceiver/                               @open-telemetry/collector-contrib-approvers @atoulme
 receiver/couchdbreceiver/                                @open-telemetry/collector-contrib-approvers @djaglowski
 receiver/dockerstatsreceiver/                            @open-telemetry/collector-contrib-approvers @rmfitzpatrick @jamesmoessis
+receiver/datadogreceiver/                                @open-telemetry/collector-contrib-approvers @boostchicken
```

Comment:
@MovieStoreGuy are you willing to sponsor/be a code owner as a contrib approver?

Reply:
Happy to

Reply:
It should also include myself and @gouthamve, as per https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1852#issuecomment-1403663819

Reply:
Done